nir/src/main/scala/scala/scalanative/nir/Defns.scala

```diff
@@ -2,33 +2,78 @@ package scala.scalanative
 package nir
 import scala.scalanative.nir.Defn.Define
 
+/** A definition in NIR.
+ *
+ *  Programs in NIR are represented as a sequence of definitions denoting types,
+ *  methods and fields. Definitions fall into two categories:
+ *
+ *    - Top-level definitions: these represent classes, modules, or traits.
+ *      Classes and modules inherit from a single parent with the exception of
+ *      `java.lang.Object`, which sits at the top of the hierarchy. They may
+ *      additionally implement traits.
+ *    - Member definitions: these represent fields or methods.
+ *
+ *  Definitions may carry attributes providing further information about their
+ *  semantics.
+ */
 sealed abstract class Defn {
+
+  /** Returns the name of the definition. */
   def name: Global
+
+  /** Returns the attributes of the definition. */
   def attrs: Attrs
+
+  /** Returns the site in the program sources corresponding to the definition.
+   */
   def pos: Position
-  final def show: String = nir.Show(this)
+
+  /** Returns a textual representation of `this`. */
+  final def show: String =
+    nir.Show(this)
+
+  /** Returns `true` iff `this` is considered as an entry point by reachability
+   *  analysis.
+   */
   final def isEntryPoint = this match {
     case Define(attrs, Global.Member(_, sig), _, _, _) =>
       sig.isClinit || attrs.isExtern
     case _ => false
   }
+
 }
 
 object Defn {
-  // low-level
-  final case class Var(attrs: Attrs, name: Global.Member, ty: Type, rhs: Val)(
-      implicit val pos: Position
-  ) extends Defn
-  final case class Const(attrs: Attrs, name: Global.Member, ty: Type, rhs: Val)(
-      implicit val pos: Position
-  ) extends Defn
+
+  // === Member definitions ==== //
+
+  /** A variable definition corresponding to a field in class or module. */
+  final case class Var(
+      attrs: Attrs,
+      name: Global.Member,
+      ty: Type,
+      rhs: Val
+  )(implicit val pos: Position)
+      extends Defn
+
+  /** A constant value. */
```

Comment:
A constant, but rather in a meaning of unique read-only instance of given type. It's important not not confuse them with constant literal, which exists only as a Val.